src/backend/utils/adt/agtype_util.c

```diff
@@ -356,6 +360,16 @@ int compare_agtype_containers_orderability(agtype_container *a,
                 break;
             }
 
+            /*Correction step because AGTV_ARRAY might be there just because of the container type*/
+            if(va.type == AGTV_ARRAY && vb.type == AGTV_OBJECT)
+            {
+                ra = agtype_iterator_next(&ita, &va, false);
+            }
+            else if(va.type == AGTV_OBJECT && vb.type == AGTV_ARRAY)
+            {
+                rb = agtype_iterator_next(&itb, &vb, false);
+            }
```

Comment:
What about a final else? What if the if and else if don't apply?

Reply:
In the final case, the rest of the function executes as it is. Should I replace the if, else if with 2 ifs?
Because they refer to very special cases. There are 2 ifs only because its possible that the first one is array and second one is object, or second one is array and first one is object. Which would be the preferred code structure in this situation?

Reply:
So long as the final else isn't necessary, then leaving it out is okay. I just wanted to make sure you checked that.

Reply:
I have added comments to explain the absence of last else.